src/DIRAC/MonitoringSystem/Client/MonitoringReporter.py

```diff
@@ -164,6 +165,18 @@ def commit(self):
             self.__documentLock.release()
         return S_OK(recordSent)
 
+    def delayedCommit(self):
+        """
+        If needed start a timer that will run the commit later
+        allowing to send more registers at once (reduces overheads).
+        """
+
+        if not self.__commitTimer.is_alive():
```

Comment:
Should this do something to make sure `self.commit` is always called, even if the python process shuts down?

Reply:
I mainly mirrored what is done for Accounting... What would you suggest?